### ISSUE TITLE
about SRResNet

### ISSUE THREAD

**Issue by denghuiru** [ShenghaiRong/caffe_srgan#1]
Hi, happy to see that you are trying to implement SRGAN using caffe, I am wonder whether you have implemented the PSNR of SRResNet in paper proposed.

**Comment by ShenghaiRong** [comment]
You can see the benchmarks in  readme.md . The PSNR of SRResNet-MSE in paper proposed is 32.05, but right now my best result is 31.26. I'm still trying to modify the params and finetuning the networks.

**Comment by denghuiru** [comment]
@ShenghaiRong 
I have trying to add sub-pixel layer as you said, however, after modify the reshape_layer.cpp && .hpp, it still can not sucessfully make caffe, I am wonder if you have also modified other codes. Would you paste it into your code?

**Comment by ShenghaiRong** [comment]
Did you modify the caffe.proto? Add `optional int32 pixelshuffler = 4 [default = 1];` to your reshape_param in caffe.proto. You can refference my caffe.proto .

**Comment by denghuiru** [comment]
@ShenghaiRong 
Hi, I am wonder have you trained the SRGAN-MSE networks, and have you got some good results?

**Comment by ShenghaiRong** [comment]
Yes，I have trained them. But the results were not good.The results showed that the impacts of GAN were not obvious.

**Comment by denghuiru** [comment]
@ShenghaiRong 
The SRGAN-MSE is not very obviously if you test the results on Set5 just as the paper said, Have you test the corresponding PSNR values,dose it match with the paper proposed?

**Comment by ShenghaiRong** [comment]
Have you trained the SRGAN-MSE? My PSNR of the SRGAN-MSE is 31.24 and the SSIM is 0.5188.

**Comment by denghuiru** [comment]
@ShenghaiRong 
yeah, I have also try to implement the SRGAN-MSE, but ,however, the PSNR dose not decrease so quickly, I think, because the SRGAN-MSE results are not so obviously, so if we can reach the psnr proposed in the paper,that will demonstrate the this training is correctly.

**Comment by Jee-King** [comment]
Hello,I have modified the caffe.proto, add optional int32 pixelshuffler = 4 [default = 1].but it have a error: Error parsing text-format caffe.NetParameter: 702:20: Message type "caffe.ReshapeParameter" has no field named "pixelshuffler".
How to solve it?

**Comment by denghuiru** [comment]
@ShenghaiRong 
hi, i have saw that you upload the segan_train.py, does this is the new code to train the srgan? And have you verify the correctness of this code?

**Comment by ShenghaiRong** [comment]
Right now I have uploaded the file of pycaffe_srgan. I'm still training the new version of srgan. But I'm busy with my dissertation. Maybe you can help me verify the correctness of this code :)

**Comment by pangyunping** [comment]
@Ji-qing  You should edit your caffe.proto,add   
optional int32 pixelshuffler = 4 [default = 1]; 
to your function ReshapeParameter {

**Comment by Nasrinimp** [comment]
Hi,

Are the versions of matlab and ubuntu important? Because with ubuntu 16.04 and matlab 2014b (also 2014a) I get this error:

Invalid MEX-file '/home/ni/Downloads/caffe/matlab/+caffe/private/caffe_.mexa64':
/home/ni/Downloads/matlab/bin/glnxa64/../../sys/os/glnxa64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found
(required by /home/ni/Downloads/caffe/matlab/+caffe/private/caffe_.mexa64)

Error in caffe.set_mode_gpu (line 5)
caffe_('set_mode_gpu');

Error in test_DRRN_B1U9 (line 65)
caffe.set_mode_gpu(); % for GPU

Thank you.

**Comment by Imanpour** [comment]
Hi,

Just for adding pixel shuffling layer in caffe what shold we do? I mean the exact changes in Caffe files for using this layer.

Thanks

**Comment by ShenghaiRong** [comment]
you can see the **Implementation Details 1**：replace your  reshape.hpp , cpp.

**Comment by ShenghaiRong** [comment]
@Imanpour you can see the Implementation Details 1：replace your reshape.hpp , cpp.

**Comment by ghost** [comment]
Thanks for your answer. I did that.
Does this work for any up scaling factor?

**Comment by ShenghaiRong** [comment]
Maybe ...  :)

**Comment by ShenghaiRong** [comment]
@Imanpour you can see the **Implementation Details 1** in the readme.md ：replace your reshape.hpp , cpp.
